### PR TITLE
Upgrade yq from v3.x to v4.x

### LINKED ISSUE #719: Upgrade yq from v3.x to v4.x
Some example scripts still use an older version of `yq`. We should upgrade this.

Please see here the upgrade guide https://mikefarah.gitbook.io/yq/upgrading-from-v3

A list of files we need to update:

```bash
➜  fabric-private-chaincode git:(main) ✗ rg "yq " .
./README.md
273:* yq v3.x (newer versions, v4.x and higher, are currently *not* supported!)

./samples/deployment/test-network/setup.sh
47:yq m -i -a=append ${CORE_PATH} core_ext.yaml
83:  yq w -i ${DOCKER_COMPOSE_TEST_NET} "services.\"$p\".volumes[+]" "${FPC_PATH_HOST}:/opt/gopath/src/github.com/hyperledger/fabric-private-chaincode"
84:  yq w -i ${DOCKER_COMPOSE_TEST_NET} "services.\"$p\".volumes[+]" "${FABRIC_SAMPLES_HOST}/config/core.yaml:/etc/hyperledger/fabric/core.yaml"

./samples/deployment/fabric-smart-client/the-simple-testing-network/env.sh
35:MSP_ID=$(yq r $GATEWAY_CONFIG organizations.${ORG^}.mspid)
36:PEER_ID=$(yq r $GATEWAY_CONFIG organizations.${ORG^}.peers[0])
37:ADDR=$(yq r $GATEWAY_CONFIG peers.\"${PEER_ID}\".url | sed 's/grpcs:\/\///')

./samples/deployment/test-network/update-connection.sh
50:  yq w -i ${CONNECTIONS_PATH} organizations.${org^}.cryptoPath ${ORG_PATH}/msp
51:  yq w -i ${CONNECTIONS_PATH} organizations.${org^}.users.${user}.cert.path "${CERTS[0]}"
52:  yq w -i ${CONNECTIONS_PATH} organizations.${org^}.users.${user}.key.path "${KEYS[0]}"
55:  yq m -i ${CONNECTIONS_PATH} - <<EOF
78:  yq r --printMode pv "${CONNECTIONS_PATH}" peers >> "${tmp_dir}/peers-${org}.yaml"
82:yq m "${tmp_dir}"/peers-*.yaml >> "${tmp_dir}/peers.yaml"
83:yq v "${tmp_dir}/peers.yaml"
89:  yq m -i "${CONNECTIONS_PATH}" "${tmp_dir}/peers.yaml"
90:  yq v "${CONNECTIONS_PATH}"

./samples/deployment/test-network/fabric-samples/test-network-nano-bash/README.md
38:yq -i 'del(.chaincode.externalBuilders) | .chaincode.externalBuilders[0].name = "ccaas_builder" | .chaincode.externalBuilders[0].path = env(PWD) + "/builders/ccaas" | .chaincode.externalBuilders[0].propagateEnvironment[0] = "CHAINCODE_AS_A_SERVICE_BUILDER_CONFIG"' config/core.yaml

./TODO.md
13:- upgrade yq (our old dep explodes alot)
./README.md
276:  GO111MODULE=on go get github.com/mikefarah/yq/v3

./utils/docker/dev_peer_cc-builder/Dockerfile
185:    go get github.com/mikefarah/yq/v3

./utils/docker/base-dev/Dockerfile
89:  && go install github.com/mikefarah/yq/v3@latest
```

### PR DESCRIPTION
**Pull request by Deepjyoti-Sarmah**
Some example scripts still use an older version of yq. To run the example scripts properly newer version of yq is added.

**What this PR does / why we need it**: Upgrades the yq V3.x to yq v4.x

**Which issue(s) this PR fixes**:
- upgrades the yq from v3.x to v4.x 
- updates the yq codes in the list of files below :
 - `./samples/deployment/test-network/setup.sh`
 - `./samples/deployment/fabric-smart-client/the-simple-testing-network/env.sh`
 - `./samples/deployment/test-network/update-connection.sh`
 - `./samples/deployment/test-network/fabric-samples/test-network-nano-bash/README.md`
 -  `./README.md`
 - `./utils/docker/dev_peer_cc-builder/Dockerfile`
 - `./utils/docker/base-dev/Dockerfile`

Fixes #719

### CODE REVIEW

File: utils/docker/base-dev/Dockerfile [hunk 1]
```diff
@@ -86,7 +86,7 @@ RUN go install golang.org/x/tools/cmd/goimports@latest \
   && go install github.com/maxbrunsfeld/counterfeiter/v6@latest \
   && go install honnef.co/go/tools/cmd/staticcheck@2023.1.3 \
   && go install github.com/client9/misspell/cmd/misspell@latest \
-  && go install github.com/mikefarah/yq/v3@latest
+  && go get github.com/mikefarah/yq/v4@latest
```
Comment (mbrandenburger) [inline utils/docker/base-dev/Dockerfile:89]: I believe this should be `go install ...`

Comment (Deepjyoti-Sarmah) [reply utils/docker/base-dev/Dockerfile]: I thank you pointing out the error and fixing it.

**Comment by mbrandenburger** [comment]
@chenchanglew may I ask you to look at this PR as well and test fpc samples?

File: samples/deployment/fabric-smart-client/the-simple-testing-network/env.sh [hunk 2]
```diff
@@ -32,9 +32,9 @@ if [[ ! -f "$GATEWAY_CONFIG" ]]; then
     exit 1
 fi
 
-MSP_ID=$(yq r $GATEWAY_CONFIG organizations.${ORG^}.mspid)
-PEER_ID=$(yq r $GATEWAY_CONFIG organizations.${ORG^}.peers[0])
-ADDR=$(yq r $GATEWAY_CONFIG peers.\"${PEER_ID}\".url | sed 's/grpcs:\/\///')
+MSP_ID=$(yq e ".organizations.${ORG^}.mspid" $GATEWAY_CONFIG)
+PEER_ID=$(yq e ".organizations.${ORG^}.peers[0]" $GATEWAY_CONFIG)
+ADDR=$(yq e ".peers.\"${PEER_ID}\".url" $GATEWAY_CONFIG | sed 's/grpcs:\/\///')
```
Comment (mbrandenburger) [inline samples/deployment/fabric-smart-client/the-simple-testing-network/env.sh:37]:
```suggestion
MSP_ID=$(yq ".organizations.${ORG^}.mspid" "$GATEWAY_CONFIG")
PEER_ID=$(yq ".organizations.${ORG^}.peers[0]" "$GATEWAY_CONFIG")
ADDR=$(yq ".peers.\"${PEER_ID}\".url" "$GATEWAY_CONFIG" | sed 's/grpcs:\/\///')
```

**Review by mbrandenburger — COMMENTED** [review]
Thanks @Deepjyoti-Sarmah for this PR! I tested it almost worked. Here some changes.

**Comment by Deepjyoti-Sarmah** [comment]
Hey @mbrandenburger thank you for the suggestion. I have made the required changes

File: samples/deployment/test-network/setup.sh [hunk 3]
```diff
@@ -44,7 +44,7 @@ fi
 echo "Adding FPC external builder to core.yaml"
 # Create a backup copy of `core.yaml` first and always work from there.
 backup ${CORE_PATH}
-yq m -i -a=append ${CORE_PATH} core_ext.yaml
+yq eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' ${CORE_PATH} core_ext.yaml -i
```
Comment (mbrandenburger) [inline samples/deployment/test-network/setup.sh:47]:
```suggestion
yq eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' -i ${CORE_PATH} core_ext.yaml
```

File: samples/deployment/test-network/update-connection.sh [hunk 4]
```diff
@@ -75,19 +76,19 @@ entityMatchers:
...
   # fetch all peers from connections
-  yq r --printMode pv "${CONNECTIONS_PATH}" peers >> "${tmp_dir}/peers-${org}.yaml"
+  yq ".peers" "${CONNECTIONS_PATH}" >> "${tmp_dir}/peers-${org}.yaml"
 done
 
 # consolidate all collected peers in a single peers.yaml
-yq m "${tmp_dir}"/peers-*.yaml >> "${tmp_dir}/peers.yaml"
-yq v "${tmp_dir}/peers.yaml"
+yq eval-all '. as $item ireduce ({}; . * $item )' ${tmp_dir}/peers-*.yaml >> "${tmp_dir}/peers.yaml"
+yq 'true' "${tmp_dir}/peers-${org}.yaml" > /dev/null
```
Comment (mbrandenburger) [inline samples/deployment/test-network/update-connection.sh:84]:
```suggestion
yq 'true' "${tmp_dir}/peers.yaml" > /dev/null
```

**Pull request merged**